### DISCUSSION TITLE
Ability to keep table filters and ordering upon refresh

### DISCUSSION THREAD

**Discussion by tallpress** [grafana/grafana#38212] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
The ability to keep table filters and ordering upon refreshing the table.

**Why is this needed**:
When browsing tabular data with a scheduled refresh rate (5s, 10s, 1min, etc.), it isn't possible to use column filtering/ordering as upon each refresh the column's filter/ordering is cleared.

It would be nice to be able to browse a subset of the data (by using filters and ordering) but also have the data refresh at a selected interval without having to reselect the filters and ordering.

**Comment by torkelo** [comment]
Sort order is preserved when I test

**Comment by tallpress** [comment]
I upgraded to 7.3.7 and have noticed the sort order is preserved, however filters are still lost upon refresh. Do your filter preferences persist after a refresh?

**Comment by ivanahuckova** [comment]
Yes, as you are saying, filters are lost upon refresh. Table filter was intended for more ad-hoc use, but I agree that removing of filter on each refresh is not useful. 
For now, for more permanent filtering, you can use Filter by value transformation. 
![image](https://user-images.githubusercontent.com/30407135/106019332-d184df00-60c2-11eb-8a9d-3d21c35fa705.png)

**Comment by ivanahuckova** [comment]
@hugohaggmark I wasn't sure if I should add `type/bug` or `type/feature-request` label. It currently works as intended, but I agree that if we just send a new query (using refresh) and not refresh the whole page, we shouldn't lose the filter.

**Comment by tallpress** [comment]
Thanks @ivanahuckova, the feature would be great to have in the future. I will try using Transforms in the meantime.

**Comment by hugohaggmark** [comment]
This is by design, table filters are transient.

**Comment by tallpress** [comment]
Hi @hugohaggmark, does this mean this feature has been "dropped" from the backlog?

**Comment by hugohaggmark** [comment]
@tallpress it was never on the backlog but it was assigned to a project board incorrectly.

**Comment by treuherz** [comment]
I'd like to push back on the idea that this isn't a usability bug, even if it's the intended behaviour. Most features of a dashboard can be configured in a panel's options, or on the front page of the dashboard, when moving things around, and unless saved will not be there when the page is reloaded or revisited, unless explicitly saved. I think this is pretty well understood by users, but the behaviour of filters doesn't fit into this paradigm at all. 

If you use dashboards for live monitoring, and have scheduled refresh running, then a few seconds after you set a filter on a table it will disappear from in front of you without any interaction from you, or any indication as to what happened. Especially since sort order persists in tables, filters being aggressively reset definitely feels unexpected.

**Comment by tlifschitz** [comment]
I agree with @treuherz. Please reconsider implementing this feature.

**Comment by na-am** [comment]
me2, that makes no sense keeping sort orders/removing filters after refreshing the dashboard

**Comment by atomlab** [comment]
The same bug in v8.1.2.

**Comment by hugohaggmark** [comment]
Please add upvotes on this issue instead of comments if you think this is a feature we should reconsider. Thank you 🙏

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Reply by ket000** [reply]
Please consider this in future versions. I routinely see a common dashboard shared by others and need to just filter for items that interest to me. Losing filter on each refresh is annoying.

**Reply by Tafsiralam** [reply]
Agree with @ket000. Please consider implementing this feature.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I tested this in my local instance and the original reported issue, where sort and filter on the table panel wasn't persisted on the automatic refresh, no longer occurs. If there is an aspect of this feature that was not addressed, please create a new issue detailing what you'd like to see done, and it will be triaged and prioritized along with the team's other work.
